src/main/java/com/lafaspot/pop/session/PopSession.java

```diff
@@ -162,10 +184,10 @@ public void operationComplete(final Future<? super Void> future) throws Exceptio
      * @throws PopException on failure
      */
     public PopFuture<PopCommandResponse> disconnect() throws PopException {
-    	final State state = stateRef.get();
-    	if (state == State.NULL) {
+        final State state = stateRef.get();
+        if (state == State.NULL) {
```

Comment:
check for connected state before disconnect

Reply:
Check if state is not CONNECTED and throw.